tools/cockpit.spec

```diff
@@ -97,14 +97,14 @@ Requires: %{name}-system = %{version}-%{release}
 # Optional components (for f24 we use soft deps)
 %if 0%{?fedora} >= 24 || 0%{?rhel} >= 8
 Recommends: %{name}-dashboard = %{version}-%{release}
-Recommends: %{name}-networkmanager = %{version}-%{release}
-Recommends: %{name}-storaged = %{version}-%{release}
+Recommends: (%{name}-networkmanager = %{version}-%{release} if NetworkManager)
+Recommends: (%{name}-storaged = %{version}-%{release} if udisks2)
```

Comment:
Note that this is not strictly correct for Fedora 26, as there it's still `storaged`. But we don't release there any more, and I didn't want to make these dependencies more complex than necessary.

Reply:
in theory you could make `Rec: (foo if (bar or baz))`. I think this should work.

Reply:
Right, I thought about that, but I think this (1) makes the dependency unnecessarily complex, and (2) introduces ambiguity. We don't actually support storaged on Fedora > 26 or the next RHEL version any more, only udisks2. So if that ever becomes a problem, I'd rather do another nested `%if` for Fedora 26. But I think it's not really necessary any more at this point.